src/main/java/io/vertx/core/http/impl/pool/Pool.java

```diff
@@ -30,16 +30,18 @@
  *
  * Pool invariants:
  * - the pool {@link #capacity} is the sum of the {@link Holder#capacity} of the {@link #available} list
- * - a connection in the {@link #available} list has its {@code Holder#capacity > 0}
+ * - a connection is in the {@link #available} list has its {@code Holder#capacity > 0}
  * - the {@link #weight} is the sum of all inflight connections {@link Holder#weight}
  *
  * A connection is delivered to a {@link Waiter} on the connection's event loop thread, the waiter must take care of
  * calling {@link io.vertx.core.impl.ContextInternal#executeFromIO} if necessary.
  *
  * Calls to the pool are synchronized on the pool to avoid race conditions and maintain its invariants. This pool can
- * be called from different threads safely (although it is not encouraged for performance reasons), since it synchronizes
- * on the pool and callbacks are asynchronous on the event loop thread of the connection without holding the pool
- * lock to avoid deadlocks. We benefit from biased locking which makes the overhead of synchronized near zero.
+ * be called from different threads safely (although it is not encouraged for performance reasons, we benefit from biased
+ * locking which makes the overhead of synchronized near zero), since it synchronizes on the pool.
+ *
+ * - acquisition success are on the event loop thread of the connection without holding the pool lock to avoid deadlocks.
+ * - acquisition failures are on an event loop too without holding the pool lock to avoid deadlocks.
```

Comment:
Maybe rephrase both sentences to:

_In order to avoid deadlocks, acquisition events (success or failure) are handled on the event loop thread of the connection without holding the pool lock._